components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/RedisSessionDataStore.java

```diff
@@ -0,0 +1,415 @@
+package org.wso2.carbon.identity.application.authentication.framework.store;
+
+import org.apache.commons.lang.SerializationUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.CarbonContext;
+import org.wso2.carbon.identity.application.common.cache.CacheEntry;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPoolConfig;
+
+import java.io.Serializable;
+import java.util.concurrent.BlockingDeque;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingDeque;
+
+public class RedisSessionDataStore extends SessionDataStore {
+
+    public static final String NANO_TIME = "nanoTime";
+    public static final String TYPE = "type";
+    public static final String EXPIRY_TIME = "expireTime";
+    public static final String TENANT_ID = "tenantId";
+    public static final String HOST = "127.0.0.1";
```

Comment:
Make these configurable

Reply:
It would be nice to keep all the constant in a separate file. That will shorten the length of the class thus easier to read.

Reply:
All other than HOST are constant values aiya. The host has to be changed as configurable.